modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/search/test/CommerceOrderIndexerTest.java

```diff
@@ -0,0 +1,279 @@
+			country.getCountryId(), _serviceContext);
+
+		return _addressLocalService.addAddress(
+			RandomTestUtil.randomString(), _user.getUserId(),
+			AccountEntry.class.getName(), _accountEntry.getAccountEntryId(),
+			country.getCountryId(), 0, region.getRegionId(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(), false,
+			RandomTestUtil.randomString(), true, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(), null,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			_serviceContext);
+	}
+
+	private void _assertSearch(
+			Hits hits, CommerceOrder... expectedCommerceOrders)
+		throws Exception {
+
+		List<CommerceOrder> actualCommerceOrders = _getCommerceOrders(hits);
+
+		long[] actualCommerceOrderIds = _getCommerceOrderIds(
+			actualCommerceOrders);
+
+		long[] expectedCommerceOrderIds = _getCommerceOrderIds(
+			Arrays.asList(expectedCommerceOrders));
+
+		Assert.assertArrayEquals(
+			expectedCommerceOrderIds, actualCommerceOrderIds);
+	}
+
+	private void _assertSearch(
+			String keywords, CommerceOrder... expectedCommerceOrders)
+		throws Exception {
+
+		SearchContext searchContext = _getSearchContext();
+
+		searchContext.setKeywords(keywords);
+
+		Hits hits = _indexer.search(searchContext);
+
+		_assertSearch(hits, expectedCommerceOrders);
+	}
+
+	private CommerceOrder _getCommerceOrder(Document document)
+		throws Exception {
+
+		long commerceOrderId = GetterUtil.getLong(
+			document.get(Field.ENTRY_CLASS_PK));
+
+		return _commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+	}
+
+	private long[] _getCommerceOrderIds(List<CommerceOrder> commerceOrders) {
+		long[] commerceOrderIds = new long[commerceOrders.size()];
+
+		for (int i = 0; i < commerceOrders.size(); i++) {
+			CommerceOrder commerceOrder = commerceOrders.get(i);
+
+			commerceOrderIds[i] = commerceOrder.getCommerceOrderId();
+		}
+
+		Arrays.sort(commerceOrderIds);
+
+		return commerceOrderIds;
+	}
+
+	private List<CommerceOrder> _getCommerceOrders(Hits hits) throws Exception {
+		Document[] documents = hits.getDocs();
+
+		List<CommerceOrder> commerceOrders = new ArrayList<>(documents.length);
+
+		for (Document document : documents) {
+			commerceOrders.add(_getCommerceOrder(document));
+		}
+
+		return commerceOrders;
+	}
+
+	private SearchContext _getSearchContext() {
+		SearchContext searchContext = new SearchContext();
+
+		searchContext.setCompanyId(_group.getCompanyId());
+		searchContext.setGroupIds(new long[] {_commerceChannel.getGroupId()});
+		searchContext.setSorts(SortFactoryUtil.getDefaultSorts());
+
+		return searchContext;
+	}
+
+	@Inject
+	private static IndexerRegistry _indexerRegistry;
+
+	@DeleteAfterTestRun
+	private AccountEntry _accountEntry;
+
+	@Inject
+	private AccountEntryLocalService _accountEntryLocalService;
+
+	@Inject
+	private AddressLocalService _addressLocalService;
+
+	@DeleteAfterTestRun
```

Comment:
not needed. There is only a test in this file so it's already removed by the dataguard